specification/attestation/data-plane/Microsoft.Attestation/stable/2020-10-01/attestation.json

```diff
@@ -0,0 +1,824 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "AttestationClient",
+    "description": "Describes the interface for the per-tenant enclave service.",
+    "version": "2020-10-01"
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{tenantBaseUrl}",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "name": "tenantBaseUrl",
+        "x-ms-parameter-location": "client",
+        "description": "The tenant name, for example https://mytenant.attest.azure.net.",
+        "required": true,
+        "type": "string",
+        "in": "path",
+        "x-ms-skip-url-encoding": true
+      }
+    ]
+  },
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/operations/policy/updatepolicy": {
+      "post": {
+        "tags": [
+          "Policy"
+        ],
+        "summary": "Accepts a new policy document and returns a JWT which expresses  used in preparation to set attestation policy.",
+        "operationId": "Policy_PrepareToSet",
+        "consumes": [
+          "text/plain"
+        ],
+        "produces": [
+          "text/plain",
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TeeKind"
+          },
+          {
+            "name": "PolicyJws",
+            "in": "body",
+            "description": "JSON Web Signature (See RFC7515) expressing the new policy",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success - Returns a JWT signed by the metadata signing key that contains the hash of the supplied policy to be set.",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "400": {
```

Comment:
By default, the status codes listed in the `responses` section are the success codes.  By listing a 400 here, the API call won't fail (e.g. throw an exception in languages that use exceptions) if a 400 is received which I don't think is what you want.

To model specific status codes as errors you must specify the [x-ms-error-response](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-error-response) extension.  Note that this is only useful for error responses that have a different schema than what's specified in your `default` clause below.  Looks like 400 is the same so it could be omitted?  401 appears to have a different schema; if this is intentional then it will need the `x-ms-error-response` extension.

Reply:
The reason for the enumerated 4xx codes was that it was in our -preview swagger and I carried it forward. If that is no longer the expectation, I can easily make that change.

Reply:
BTW, what is the standard mechanism for handling 401 errors?

Reply:
The standard behavior is that any response not in the `responses` section will be considered an error and an exception thrown (or whatever error-path mechanism is appropriate for the language).  If a schema is specified for `default` then the error response _should_ get deserialized into that type (`CloudError` here) and a matching exception type thrown (`CloudErrorException` I presume for C#).  If no schema is specified the I believe some generic error is thrown/returned that contains the response body if available.

Only if you want distinct, strongly-typed error models for different error status codes (400, 401) should you have a need to specify error status codes under `responses`.  And that also requires specifying `x-ms-error-response`.

Reply:
Ok. Looked over a bunch of other REST specs for 401 errors and it looks like most don't special case it (and there are a bunch which are really unusual).